File: src/ParserNbBet/Kush/BetResult.cs
namespace ParserNbBet.Kush;

/// <summary>
/// Result of a bet placement attempt.
/// </summary>
public sealed class BetResult
{
    public bool Success { get; }
    public bool DryRun { get; }
    public string BetType { get; }
    public double OddsNb { get; }
    public double OddsKush { get; }
    public double Ratio { get; }
    public double Stake { get; }
    public string KushEventId { get; }
    public string Reason { get; }

    public BetResult(bool success, bool dryRun, string betType,
                     double oddsNb, double oddsKush, double ratio,
                     double stake, string kushEventId, string reason)
    {
        Success = success;
        DryRun = dryRun;
        BetType = betType;
        OddsNb = oddsNb;
        OddsKush = oddsKush;
        Ratio = ratio;
        Stake = stake;
        KushEventId = kushEventId;
        Reason = reason;
    }

    public static BetResult Failed(string reason, string betType = "", string kushEventId = "") =>
        new(false, false, betType, 0, 0, 0, 0, kushEventId, reason);

    public override string ToString() =>
        DryRun
            ? $"[DRY-RUN] {BetType}: ratio={Ratio:F4} kush={OddsKush:F2} nb={OddsNb:F2} stake={Stake} — {Reason}"
            : $"[{(Success ? "OK" : "FAIL")}] {BetType}: ratio={Ratio:F4} kush={OddsKush:F2} nb={OddsNb:F2} stake={Stake} — {Reason}";
}

File: src/kush/bet_result.py
"""Bet result dataclass."""
from __future__ import annotations

from dataclasses import dataclass, field
from datetime import datetime


@dataclass
class BetResult:
    """Result of a bet placement attempt."""

    match_key: str
    event_id: str
    bet_type: str  # e.g. "П1", "П2", "X", "1X", "ТБ (2.50)"
    kf_nb: float  # NB-Bet coefficient
    kf_kush: float  # Kush coefficient
    ratio: float  # kf_kush * (1 + roi) / kf_nb
    threshold: float  # 1.10 or 1.05
    ratio_passes: bool
    placed: bool  # True if bet was actually placed (not dry-run)
    dry_run: bool
    success: bool  # True if placement succeeded (or dry-run passed)
    error: str = ""
    insufficient_funds: bool = False  # Kush rejected: not enough balance
    insufficient_funds: bool = False  # True if Kush rejected due to lack of funds
    timestamp: datetime = field(default_factory=lambda: datetime.now())
    league: str = ""
    team_home: str = ""
    team_away: str = ""

    @property
    def summary(self) -> str:
        mode = "DRY-RUN" if self.dry_run else "REAL"
        status = "OK" if self.success else f"FAIL: {self.error}"
        return (
            f"[{mode}] {self.team_home} vs {self.team_away} | "
            f"{self.bet_type} kf={self.kf_kush:.2f} "
            f"ratio={self.ratio:.3f}/{self.threshold:.2f} | {status}"
        )
